core/src/test/java/org/apache/iceberg/TestCommitReporting.java

```diff
@@ -196,14 +196,9 @@ public void addAndDeleteManifests() throws IOException {
 
     CommitReport report = reporter.lastCommitReport();
     assertThat(report).isNotNull();
-    assertThat(report.operation()).isEqualTo("append");
-    assertThat(report.snapshotId()).isEqualTo(2L);
-    assertThat(report.sequenceNumber()).isEqualTo(2L);
+    assertThat(report.operation()).isEqualTo("replace");
+    assertThat(report.snapshotId()).isEqualTo(3L);
+    assertThat(report.sequenceNumber()).isEqualTo(3L);
     assertThat(report.tableName()).isEqualTo(tableName);
-
-    CommitMetricsResult metrics = report.commitMetrics();
```

Comment:
I'd probably change this to 
```
CommitMetricsResult metrics = report.commitMetrics();
assertThat(metrics.totalDataFiles().value()).isEqualTo(2L);
assertThat(metrics.totalRecords().value()).isEqualTo(2L);
assertThat(metrics.totalFilesSizeInBytes().value()).isEqualTo(20L);
```
so that we at least check that some metrics were recorded

Reply:
I'm fine with the PR as-is. I'll follow up in a separate PR to add proper commit metrics for manifests kept/created/replaced

Reply:
@bryanck I've opened https://github.com/apache/iceberg/pull/12630 to address ^